### PR TITLE
Add region to request filtering

### PR DESCRIPTION
**Pull request by hobbitronics**
- changed autocomplete options type to regions
- added filtering properties for:
  - locality
  - sublocality
  - administrative_area_level_1
  - administrative_area_level_2
- filter requests by region if not within 100km
- remove cities from search field placeholder
Screen Shot 2021-07-07 at 11 30 57 AM

### CODE REVIEW

File: src/data/requestFiltering.js [hunk 1]
```diff
@@ -82,6 +98,17 @@ export function populateRequestFilterFrom(queryStringData, me, events) {
   }
 }
 
+function matchByRegion (location, requestLocation) {
+  if (!location || !requestLocation) return true
+  if (location.sublocality || location.locality || location.county) {
+    return isNear(location, requestLocation)
+  } else if (location.state) {
+    return requestLocation.description.includes(location.state)  || requestLocation.description.includes(location.description)
```
Comment (Schparky) [inline src/data/requestFiltering.js:106]: It is not guaranteed that the `state` property is included in the description. Try Chiang Mai, Thailand, for example. Will that break this conditional?

File: src/data/requestFiltering.js [hunk 2]
```diff
@@ -82,6 +98,17 @@ export function populateRequestFilterFrom(queryStringData, me, events) {
...
+  } else if (location.country) {
+    return requestLocation.description.includes(location.country) || requestLocation.description.includes(location.description)
```
Comment (Schparky) [inline src/data/requestFiltering.js:108]: Same as with state. It may be included, but not spelled or capitalized the same way.

**Review by Schparky — APPROVED** [review]
Approved, but with a couple caveats:
- this makes the UI search inconsistent with the backend, particularly in the processing of alerts (watches)
- it potentially makes an even longer URL with all of the places parameters encoded...is that acceptable and necessary?

File: src/data/requestFiltering.js (hunk 1 above)
Comment (hobbitronics) [reply src/data/requestFiltering.js:106]: You are correct. I made a request for 'Chiang Mai', Thailand and searching for destination 'Chiang Mai Province, Thailand' doesn't work. It seems to have a robust search by region we will need some additional property on the request.

**Comment by hobbitronics** [comment]
> Approved, but with a couple caveats:
> 
> * this makes the UI search inconsistent with the backend, particularly in the processing of alerts (watches)
> * it potentially makes an even longer URL with all of the places parameters encoded...is that acceptable and necessary?

Would it be preferable to pass the region filters to populateRequestFilterFrom() as another parameter and leave them out of the URL?

**Comment by Schparky** [comment]
> > Approved, but with a couple caveats:
> > 
> > * this makes the UI search inconsistent with the backend, particularly in the processing of alerts (watches)
> > * it potentially makes an even longer URL with all of the places parameters encoded...is that acceptable and necessary?
> 
> Would it be preferable to pass the region filters to populateRequestFilterFrom() as another parameter and leave them out of the URL?

I'm not sure. Can't remember how that works. But the reason its in the URL is to support browser navigation and bookmarking. That's a UX decision, whether or not to preserve that. I think it's OK to proceed. Just wanted to call attention to it.

**Pull request merged**